lib/ui/availability/availability_card.dart

```diff
@@ -43,24 +43,24 @@ class _AvailabilityCardState extends State<AvailabilityCard> {
   }
 
   Widget buildAvailabilityCard(List<AvailabilityModel?> data) {
-    List<Widget> locationsList = [];
-    RegExp multiPager = RegExp(r' \(\d+/\d+\)$');
-    // loop through all the models, adding each one to locationsList
-    for (AvailabilityModel? model in data) {
-      if (model != null) {
-        String curName = model.name!;
-        RegExpMatch? match = multiPager.firstMatch(curName);
-        if (match != null) {
-          curName = curName.replaceRange(match.start, match.end, '');
-        }
-        if (_availabilityDataProvider.locationViewState[curName]!) {
-          locationsList.add(AvailabilityDisplay(model: model));
-        }
+    // RegExp multiPager = RegExp(r' \(\d+/\d+\)$');
+    // Filter the models and create a list of only the valid ones
+    List<Widget> locationsList = data
```

Comment:
I can't tell if this is proper usage of functional-style iterator methods. Let's try to rework this one.

Reply:
@parthshindee let's meet on this one individually to see the best solution. Possible precedent case for the official Campus Mobile Style Guide